src/SignalR/clients/java/signalr/messagepack/src/main/java/com/microsoft/signalr/messagepack/MessagePackHubProtocol.java

```diff
@@ -121,13 +122,13 @@ public int getVersion() {
                     // Check what the last message was
                     // If it was an invocation binding failure, we have to correct the position of the buffer
                     if (hubMessages.get(hubMessages.size() - 1).getMessageType() == HubMessageType.INVOCATION_BINDING_FAILURE) {
-                        payload.position(payload.position() + (length - readBytes));
+                        ((Buffer) payload).position(payload.position() + (length - readBytes));
                     } else {
                         throw new RuntimeException(String.format("MessagePack message was length %d but claimed to be length %d.", readBytes, length));
                     }
                 }
                 unpacker.close();
-                payload.position(payload.position() + readBytes);
+                ((Buffer) payload).position(payload.position() + readBytes);
```

Comment:
Comment why you're casting to Buffer like this. Include a link to this PR.

Reply:
https://github.com/dotnet/aspnetcore/pull/26614/commits/a6998e8885e11ea8946b2abce6a3c6370ef13d14